### ISSUE TITLE
Don't use the legacy view attributes API

### ISSUE THREAD

**Issue by teosarca** [metasfresh/metasfresh-webui-frontend#1357]
Instead of using
http://w101.metasfresh.com:8081/swagger-ui.html#/view-row-attributes-rest-controller-_-old

please always use
http://w101.metasfresh.com:8081/swagger-ui.html#/view-row-attributes-rest-controller

That legacy API is scheduled to be deleted. It already caused a lot of confusion.

### Note to IT:
* check view attributes (setting, dropdown) => no errors
* briefly test creating a new document (e.g. a new sales order) => no errors

**Comment by wiadev** [comment]
@teosarca fixed calling wrong api after editing/patching on the following modal under this task we discussed last time.

![screenshot at nov 24 09-46-57](https://user-images.githubusercontent.com/18234060/33202369-91583e30-d0fc-11e7-881d-f5244a0c4a75.png)

can you give me the use cases using old apis?

**Comment by metas-lc** [comment]
how to test attributes
1. go to receipt candidates window: https://w101.metasfresh.com:8443/window/540196?viewId=540196-06faf33c94f14604ac8662311a3055a8&page=1
2. select any line (e.g. one for convenence salad)
3. select the main action (e.g. green button: 1 IFCO 6410 x 10 Stk)
4. modify some attributes while being on LU level: 
=>
when you edit gewicht brutto notice gewicht netto updates
when you update tageslot lot-nummer updates
most attributes you modify on LU level, will update on TU level as well
cu level doesn't support attributes

**Comment by wiadev** [comment]
@teosarca can you point out where I can find subentity/subentityId apis on swagger ui doc?
Also, usage of isAdvanced query param.

**Comment by teosarca** [comment]
@wiadev  let's talk when u have time. I don't understand your request :(

**Comment by wiadev** [comment]
another use case to get attributes after discussing with @teosarca 
https://w101.metasfresh.com:8443/window/540189

**Comment by wiadev** [comment]
@teosarca same issue on http://localhost:3000/window/540189?viewId=540189-d9970cd0d9764f67b6d1e44bfbfba4b7&page=1 when I select the last `includedDocument`.

**Comment by teosarca** [comment]
@wiadev 

> check out the above screenshot and could you point out why it doesn't work?
> same issue on http://localhost:3000/window/540189?viewId=540189-d9970cd0d9764f67b6d1e44bfbfba4b7&page=1 when I select the last includedDocument.

Yup that's happening because the row level `supportAttributes` property is not considered.
See task: https://github.com/metasfresh/metasfresh-webui-frontend/issues/1393

![image](https://user-images.githubusercontent.com/1244701/33389122-85e0820a-d53a-11e7-8433-5c6a2e71049a.png)

**Comment by wiadev** [comment]
@teosarca understood
since we have a new follow-up issue, can we close this issue after merge #1392 ?

**Comment by teosarca** [comment]
@wiadev  nop, the issue will be closed by the integration testing team after they will perform all tests + regression.
That's our standard procedure.

**Comment by teosarca** [comment]
but they will know about that follow-up and will not fail the task because of that.

**Comment by wiadev** [comment]
@teosarca got it.
I meant by moving to testing team.

**Comment by teosarca** [comment]
yes, ofc.
but before i will do some quick preliminary tests. One thing that i forgot to tell you and now pops in my mind, and I HOPE it's not related is about editing product attributes.

e.g.
![image](https://user-images.githubusercontent.com/1244701/33389698-6736885c-d53c-11e7-8889-4b2f37076d0f.png)

**Comment by teosarca** [comment]
update: it seems the view attributes code has nothing to do with product attributes (the ones from above). So it's fine.
@wiadev  can u also confirm this?

**Comment by wiadev** [comment]
@teosarca confirmed that viewAttributes code has nothing to do with product attributes.
It's using another getAttributeInstances() methods which is `POST /api/pattribute`

**Comment by teosarca** [comment]
@wiadev  perfect, 10x.

**Comment by teosarca** [comment]
i found out that fetching the view attributes layout was using the old API.
Pls fix it.
![image](https://user-images.githubusercontent.com/1244701/33427018-83b4975e-d5cc-11e7-9e5e-ebbf775edf67.png)

**Comment by teosarca** [comment]
found others issues too

### PATCH method is using the old API
![image](https://user-images.githubusercontent.com/1244701/33517297-4dffce70-d78a-11e7-8977-b3ab4c48135b.png)


### dropdown endpoint is using wrong URL
![image](https://user-images.githubusercontent.com/1244701/33517304-849b08b4-d78a-11e7-8155-8552f96b5202.png)

Also make sure the typeahead dropdown has the correct URL
 Pls check the correct endpoints in swagger.

**Comment by wiadev** [comment]
@teosarca where can I find the /typeahead use case?

**Comment by metas-ts** [comment]
CPed the remaining commits to `release` too

**Comment by metas-rc** [comment]
Note: IT only after https://github.com/metasfresh/metasfresh-webui-frontend/pull/1441 is properly integrated.

**Comment by metas-lc** [comment]
IT
tested attributes in orderlines, manufacturing OK
tested in hu editor from material receipt OK
made a task for a special case: https://github.com/metasfresh/metasfresh-webui-frontend/issues/1468
in hu editor, if you set an attribute from none (null) to none you get an error
